nifi-registry/nifi-registry-core/nifi-registry-web-api/src/main/java/org/apache/nifi/registry/web/security/authentication/jwt/JwtIdentityProvider.java

```diff
@@ -70,7 +71,8 @@ public AuthenticationResponse authenticate(AuthenticationRequest authenticationR
 
         try {
             final String jwtPrincipal = jwtService.getUserIdentityFromToken(jwtAuthToken);
-            return new AuthenticationResponse(jwtPrincipal, jwtPrincipal, expiration, issuer);
+            final Set<String> groups = jwtService.getUserGroupsFromToken(jwtAuthToken);
```

Comment:
This approach requires parsing the authentication token string twice, which is not very efficient. It probably requires some refactoring to do this once, and then extract the claims required.

Reply:
Refactored to improve the efficiency of token parsing / validation. I did change the method signature of a couple public methods - if you'd rather me not do that just let me know and I'll just add new methods instead of replacing the existing ones.